src/librustc/ich/hcx.rs

```diff
@@ -59,6 +61,20 @@ pub enum NodeIdHashingMode {
     HashDefPath,
 }
 
+/// The BodyResolver allows to map a BodyId to the corresponding hir::Body.
+/// We could also just store a plain reference to the hir::Crate but we want
+/// to avoid that the crate is used to get untracked access to all of the HIR.
+#[derive(Clone, Copy)]
+struct BodyResolver<'hir>(&'hir hir::Crate);
```

Comment:
Nit: I'd prefer to insert a module here, so that we can use the module API to see that the invariant is maintained. But doesn't really matter.

Reply:
I guess there is no really *useful* invariant here anyway.